gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -44,31 +44,54 @@ if SERVER then
 		ply.oldSprintProgress = ply.sprintProgress
 		ply.sprintMultiplier = bool and (1 + maxSprintMul:GetFloat()) or nil
 		ply.isSprinting = bool
-		ply.sprintTS = CurTime()
 	end)
 else
-	local function PlayerSprint(bool)
+	local lastPress = 0
+
+	local function PlayerSprint(trySprinting, bind)
 		local client = LocalPlayer()
 
-		if bool and not GetGlobalBool("ttt2_sprint_enabled", true) or not bool and not client.isSprinting then return end
+		if trySprinting and not GetGlobalBool("ttt2_sprint_enabled", true) then return end
```

Comment:
@TimGoll could you clean this up if you find some time? :) I'm unable to do so currently :/